lmdeploy/model.py

```diff
@@ -18,6 +18,31 @@ class SamplingParam:
     repetition_penalty: float = 1.0
 
 
+@dataclasses.dataclass
+class ModelConfig:
+    """parameters for chat templates."""
+    model_name: str  # determine which chat template will be applied
+    system: Optional[str] = None  # begin of the system prompt
+    meta_instruction: Optional[str] = None  # system prompt
+    eosys: Optional[str] = None  # end of the system prompt
+    user: Optional[str] = None  # begin of the user prompt
+    eoh: Optional[str] = None  # end of the user prompt
+    assistant: Optional[str] = None  # begin of the assistant prompt
+    eoa: Optional[str] = None  # end of the assistant prompt
+    capability: Optional[str] = None
+
+    @property
+    def model(self):
```

Comment:
Setting builder method as property is a little bit strange.

Reply:
This is not supposed to be called frequently. And the build process is not computationally intensive.